### PR TITLE
feat: add workflow actions

### PR DESCRIPTION
**Pull request by dpilch**
Add workflow actions basic support. I used NavigateAction as the first action, but this does not fully cover the QA for NavigateAction. Integ tests were also not done because the hooks have not been implemented yet. 

To Do in separate changes: 
* unique identifier for action name (wait for new override changes)
* add special case for parameters: model, fields
* add auth attribute support to action parameters (wait for auth attributes merged)

### CODE REVIEW

File: packages/codegen-ui-react/lib/action.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  const actionName = ActionNameMapping[Action[action as Action]];
+  if (actionName === undefined) {
+    throw new InvalidInputError(`${action} is not a valid action.`);
+  }
+  return actionName;
+}
+
+export function getComponentActions(component: StudioComponent | StudioComponentChild): ActionStudioComponentEvent[] {
+  // TODO: add unique identifier to each action
+  if (component.events) {
```
Comment (alharris-at) [inline packages/codegen-ui-react/lib/action.ts:62]: You'll miss child-only events as written. You should just treat as two separate conditionals, and create the empty array at the top.

File: packages/codegen-ui-react/lib/action.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+          ),
+        false,
+      ),
+    );
+  }
+  // TODO: determine what should be used when no parameters
+  return factory.createObjectLiteralExpression([], false);
+}
+
+export function actionParameterToExpression(parameter: StudioComponentProperty): Expression {
```
Comment (alharris-at) [inline packages/codegen-ui-react/lib/action.ts:115]: We almost certainly support a generic version of this already in the helper, right?

Comment (alharris-at) [reply packages/codegen-ui-react/lib/action.ts:115]: Or are you keeping separate to make it easier to handle the circular dependency thing here?

Comment (dpilch) [reply packages/codegen-ui-react/lib/action.ts:120]: I think we actually don't have a generic version. I wrote this portion around 2 months ago so I can't remember the exact reasoning. It looks like there is repeated logic for isFixedProperty, isBoundProperty, etc. but each adds a different spin to the output. We should refactor all to use one generic version as the base.

* https://github.com/aws-amplify/amplify-codegen-ui/blob/main/packages/codegen-ui-react/lib/react-component-render-helper.ts#L227
* https://github.com/aws-amplify/amplify-codegen-ui/blob/main/packages/codegen-ui-react/lib/react-component-render-helper.ts#L264
* https://github.com/aws-amplify/amplify-codegen-ui/blob/main/packages/codegen-ui-react/lib/react-component-render-helper.ts#L370

I think it's best to do that as a separate change though.

Comment (alharris-at) [reply packages/codegen-ui-react/lib/action.ts:120]: Yep, makes sense to me. I'll create a tracking task.

**Review by alharris-at — APPROVED** [review]

**Pull request merged**